tests/core/waiting/test_waiting_for_ipc_socket.py

```diff
@@ -2,22 +2,51 @@
 
 from flaky import flaky
 
-from geth import DevGethProcess
+from geth import (
+    DevGethProcess,
+    get_geth_version
+)
 from geth.utils.timeout import (
     Timeout,
 )
 
 
 def test_waiting_for_ipc_socket(base_dir):
-    with DevGethProcess('testing', base_dir=base_dir) as geth:
-        assert geth.is_running
-        geth.wait_for_ipc(timeout=60)
+    version = get_geth_version()
+
+    if version.major == 1:
+        if version.minor == 9:
+            with DevGethProcess('testing',
+                                base_dir=base_dir,
+                                overrides={'allow_insecure_unlock': True}) as geth:
+                assert geth.is_running
+                geth.wait_for_ipc(timeout=60)
+        elif version.minor == 7 or version.minor == 8:
+            with DevGethProcess('testing',
+                                base_dir=base_dir) as geth:
+                assert geth.is_running
+                geth.wait_for_ipc(timeout=60)
+    else:
+        assert False, "Unsupported geth version"
 
 
 @flaky(max_runs=3)
 def test_timeout_waiting_for_ipc_socket(base_dir):
-    with DevGethProcess('testing', base_dir=base_dir) as geth:
-        assert geth.is_running
-        with pytest.raises(Timeout):
-            geth.wait_for_ipc(timeout=0.1)
-        assert geth.is_running
+    version = get_geth_version()
+
+    if version.major == 1:
+        if version.minor == 9:
+            with DevGethProcess('testing',
+                                base_dir=base_dir,
+                                overrides={'allow_insecure_unlock': True}) as geth:
```

Comment:
It's a little unfortunate that this logic is repeated.  Potentially something like `_amend_geth_kwargs(base_dir=...)` that would add the `allow_insecure_unlock` conditionally and raise the error if the version isn't supported...